apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -1290,29 +1297,23 @@ extract_connector_id(Id) when is_binary(Id) ->
             <<"connector:", ConnectorType/binary, ":", ConnectorName/binary>>;
         _ ->
             Id
-    end;
-extract_connector_id(Id) ->
-    Id.
-
-is_channel_id(Id) ->
-    extract_connector_id(Id) =/= Id.
+    end.
 
 %% Check if channel is installed in the connector state.
 %% There is no need to query the conncector if the channel is not
 %% installed as the query will fail anyway.
-pre_query_channel_check({Id, _} = _Request, Channels, QueryOpts) when
-    is_map_key(Id, Channels)
-->
-    ChannelStatus = maps:get(Id, Channels),
-    case emqx_resource_manager:channel_status_is_channel_added(ChannelStatus) of
+pre_query_channel_check(Id, {Id, _} = _Request, ChanSt, QueryOpts) ->
+    case emqx_resource_manager:channel_status_is_channel_added(ChanSt) of
         true ->
             ok;
         false ->
             error_if_channel_is_not_installed(Id, QueryOpts)
     end;
-pre_query_channel_check({Id, _} = _Request, _Channels, QueryOpts) ->
+pre_query_channel_check(Id, {Id, _} = _Request, ?NO_CHANNEL, QueryOpts) ->
```

Comment:
This clause should be before the previous, since it's more specific on `ChanSt`

Reply:
this clause is not needed.